packages/upswyng-server/src/routes/api/nextSchedule.ts

```diff
@@ -0,0 +1,100 @@
+import { ResourceSchedule, TScheduleItem } from "@upswyng/upswyng-core";
+import nextScheduleCache, {
+  SchedulePeriod,
+} from "../../utility/nextScheduleCache";
+import { ObjectId } from "bson";
+import Resource from "../../models/Resource";
+import moment from "moment";
+
+const cache = nextScheduleCache();
+
+const scheduleItemToPeriod = (item: TScheduleItem): SchedulePeriod => {
+  const dateFormat = "YYYY MM DD";
+  const dateTimeFormat = `${dateFormat} HH:mm`;
+  const itemDateString = moment(item.recurrenceRule.all()[0]).format(
+    dateFormat
+  );
+  const periodOpen = moment(
+    `${itemDateString} ${item.fromTime.value}`,
+    dateTimeFormat
+  );
+  const periodClose = moment(
+    `${itemDateString} ${item.toTime.value}`,
+    dateTimeFormat
+  );
+
+  return {
+    open: periodOpen.toDate(),
+    close: periodClose.toDate(),
+  };
+};
+
+const getNextSchedulePeriod = async (id: string) => {
+  const { schedule } = await Resource.getByResourceId(
+    ObjectId.createFromHexString(id)
+  );
+  const scheduleItems = ResourceSchedule.parse(schedule).getItems();
+
+  return scheduleItems.reduce(
+    (nextSchedule: SchedulePeriod | null, currentSchedule: TScheduleItem) => {
+      const currentPeriod = scheduleItemToPeriod(currentSchedule);
+      if (!nextSchedule) {
+        return currentPeriod;
+      }
+
+      const currentPeriodIsSooner =
+        currentPeriod.open.getTime() < nextSchedule.open.getTime();
+
+      return currentPeriodIsSooner ? currentPeriod : nextSchedule;
+    },
+    null
+  );
+};
+
+export const get = async (req, res, _next) => {
+  const { resourceIds } = req.query;
+
+  if (!resourceIds) {
+    res.writeHead(400, {
+      "Content-Type": "application/json",
+    });
+
+    res.end(
+      JSON.stringify({
+        message: "Please provide resource ID(s)",
+      })
+    );
+  } else {
+    const ids = resourceIds.split(",");
```

Comment:
I haven't seen the fetch code for this, but the way I would do it would be send JSON as the body that looks like:
```
{
  resourceIds: ["adnfwoiaofo2i3q", "aksdifjwepiuh2p"]
}
```
There is already middleware in the server that runs `JSON.parse` on the body, so `resourceIds` will end up being `string[]`

Reply:
[updated](https://github.com/CodeForBoulder/upswyng/pull/265/commits/dd88d89455432239b3cc9c7023e8110e3adfed83)